### ISSUE TITLE
StringHelper::truncate returns a string with more space

### ISSUE THREAD

**Issue by kk8686** [yiisoft/yii2#12495]
```php
echo \yii\helpers\StringHelper::truncate('我<span>是a中</span>国人', 4, '...', 'UTF-8', true);
```

When I set param of `$asHtml` are true, the result is: "`我 <span>是a </span>...`"

But I'm expect is: "`我<span>是a中</span>...`"

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.9
| PHP version      |7.0.1
| Operating system |windows 10

**Comment by samdark** [comment]
Seems like unicode issues to me.

1. Have you tried code from master?
2. Any ideas on fixing it?

**Comment by kk8686** [comment]
I'm sorry I don't have tried to code from master, bacause I no any ideas

**Comment by samdark** [comment]
OK, I'll handle it.

**Comment by SilverFire** [comment]
That's not because of UTF-8 but because of adding spaces after tokens [here](https://github.com/yiisoft/yii2/blob/master/framework/helpers/BaseStringHelper.php#L170)

**Comment by kk8686** [comment]
@samdark Thanks

**Comment by TerraSkye** [comment]
Isnt this the cause of half widht characters? In the pull ruest for console::table the widh was also off.

**Comment by SilverFire** [comment]
> Isnt this the cause of half widht characters?

No, it's not

**Comment by zelenin** [comment]
Tests has extra spaces too https://github.com/yiisoft/yii2/blob/master/tests/framework/helpers/StringHelperTest.php#L110 that's funny.
@Alex-Code ping
https://github.com/yiisoft/yii2/pull/6207

**Comment by Alex-Code** [comment]
I can't remember why I added the extra space there, not sure if it was more for truncating words within HTML that spaces would get lost.
I'll do some more tests to see if that space can be removed.

**Comment by Alex-Code** [comment]
The main issue is with spaces getting removed by the `truncate` and `truncateWords` methods.
I've updated the `truncateHtml` method and added new tests which all pass.

**Comment by SilverFire** [comment]
Resolved by 6b88f2f
